vignettes/queries_dataset.Rmd

```diff
@@ -0,0 +1,179 @@
+    
+    
+    Note: The type (numeric or character) of the variable defined in `TERM_LEVEL` is checked in `dataset`. If the variable is a character variable (e.g. `AEDECOD`), it is expected that `TERM_NAME` is populated, if it is a numeric variable (e.g. `AEBDSYCD`), it is expected that `TERM_ID` is populated, otherwise an error is issued.
+    
+
+### Example
+
+In this example, 1 standard MedDRA query (`VAR_PREFIX = "SMQ01"`) and 1 customized query (`VAR_PREFIX = "CQ02"`) are  defined to analyze the adverse events. 
+
+  + The standard MedDRA query variable `SMQ01NAM` [`VAR_PREFIX`] will be populated with "Standard Query 1" [`QUERY_NAME`] if any preferred term (`AEDECOD`) [`TERM_LEVEL`] in `dataset` is equal to "AE1" or "AE2" [`TERM_NAME`]
+  
+  + The customized query (`CQ02NAM`) [`VAR_PREFIX`] will be populated with "Query 2" [`QUERY_NAME`] if any Low Level Term Code (`AELLTCD`)  [`TERM_LEVEL`] in `dataset` is equal to 10  [`TERM_ID`] or any preferred term (`AEDECOD`) [`TERM_LEVEL`] in `dataset` is equal to "AE4" [`TERM_NAME`].
+  
+  **`dataset_queries = ds_query`: **
+  
+VAR_PREFIX | QUERY_NAME | TERM_LEVEL | TERM_NAME | TERM_ID |
+------- | ----- | ------ | ----- | ----- | -----
+SMQ01| Standard Query 1 | AEDECOD | AE1 | 
+SMQ01| Standard Query 1 | AEDECOD | AE2 | 
+CQ02| Query 2 | AELLTCD | | 10
+CQ02| Query 2 | AEDECOD | AE4| 
+
+  **`dataset = ae`: **
+  
+USUBJID | AEDECOD | AELLTCD 
+------- | ----- | ------ 
+0001 | AE1 | 101 | 
+0001 | AE3 | 10| | 
+0001 | AE4 | 120 | 
+0001 | AE5 | 130 |
+
+
+  **output dataset from `derive_vars_query(dataset = ae, dataset_queries = ds_query)`:**
+
+USUBJID | AEDECOD | AELLTCD | SMQ01NAM |CQ02NAM
+------- | ----- | ------ | ----- | ----- 
+0001 | AE1 | 101 | Standard Query 1 |
+0001 | AE3 | 10| | Query 2
+0001 | AE4 | 120 |  | Query 2
+0001 | AE5 | 130 |  |
+
+
+  Subject 0001 has 1 event meeting the Standard Query 1 criteria (`AEDECOD = "AE1"`) and 2 events meeting the customized query (`AELLTCD = 10` and `AEDECOD = "AE4"`). 
+
+
+## Optional Content
+
+
+When standardized MedDRA Queries are added to the dataset, it is expected that the name of the query (`ABCzzNAM`) is populated along with its number code (`ABCzzCD`),  and its Broad or Narrow scope (`ABCzzSC`).
+
+The following variables can be added to `queries_datset` to derive this information.
+
+
+#### Input
+
+  + `QUERY_ID` must be an integer. 
+  
+  + `QUERY_SCOPE` must be a character string. Possible values are: "BROAD", "NARROW" or `NA`. 
+  
+  + `QUERY_SCOPE_NUM` must be an integer. Possible values are: `1`, `2` or `NA`. 
+
+#### Output
+
+  + `QUERY_ID`, `QUERY_SCOPE` and `QUERY_SCOPE_NUM` will be used in the same way as `QUERY_NAME` [(see here)](#query_name) and will help in the creation of the `ABCzzCD`, `ABCzzSC` and `ABCzzSCN` variables.
+  
+## Output Variables
```

Comment:
All these headings should be level 3, i.e. `###`.

Reply:
Updated